core/storage/couchStorage.go

```diff
@@ -0,0 +1,1801 @@
+func (store *CouchStorage) RetrieveTimeOnServer() (time.Time, error) {
+	return time.Now(), nil
+}
+
+// StoreOrgToMessagingGroup inserts organization to messaging groups table
+func (store *CouchStorage) StoreOrgToMessagingGroup(orgID string, messagingGroup string) common.SyncServiceError {
+
+	existingObject := &couchMessagingGroupObject{}
+	if err := store.getOne(orgID, existingObject); err != nil {
+		if err != notFound {
+			return err
+		}
+		existingObject = nil
+	}
+
+	newObject := couchMessagingGroupObject{ID: orgID, GroupName: messagingGroup}
+	if existingObject != nil {
+		newObject.Rev = existingObject.Rev
+	}
+	err := store.upsertObject(orgID, newObject)
+	if err != nil {
+		return &Error{fmt.Sprintf("Failed to store organization's messaging group. Error: %s.", err)}
+	}
+	return nil
+}
+
+// DeleteOrgToMessagingGroup deletes organization from messaging groups table
+func (store *CouchStorage) DeleteOrgToMessagingGroup(orgID string) common.SyncServiceError {
+	if err := store.deleteObject(orgID); err != nil && err != notFound {
+		return err
+	}
+	return nil
+}
+
+// RetrieveMessagingGroup retrieves messaging group for organization
+func (store *CouchStorage) RetrieveMessagingGroup(orgID string) (string, common.SyncServiceError) {
+	result := couchMessagingGroupObject{}
+	if err := store.getOne(orgID, &result); err != nil {
+		if err != notFound {
+			return "", err
+		}
+		return "", nil
+	}
+	return result.GroupName, nil
+}
+
+// RetrieveUpdatedMessagingGroups retrieves messaging groups that were updated after the specified time
+func (store *CouchStorage) RetrieveUpdatedMessagingGroups(time time.Time) ([]common.MessagingGroup,
+	common.SyncServiceError) {
+	result := []couchMessagingGroupObject{}
+	if err := store.findAll(map[string]interface{}{"selector": map[string]interface{}{"last-update": map[string]interface{}{"$gte": time}}}, &result); err != nil {
+		return nil, err
+	}
+	groups := make([]common.MessagingGroup, 0)
+	for _, group := range result {
+		groups = append(groups, common.MessagingGroup{OrgID: group.ID, GroupName: group.GroupName})
+	}
+	return groups, nil
+}
+
+// DeleteOrganization cleans up the storage from all the records associated with the organization
+func (store *CouchStorage) DeleteOrganization(orgID string) common.SyncServiceError {
+	if err := store.DeleteOrgToMessagingGroup(orgID); err != nil {
+		return err
+	}
+
+	if err := store.deleteAllDestinationObjects(map[string]interface{}{"selector": map[string]interface{}{"destination.destinationOrgID": orgID}}); err != nil && err != notFound {
+		return &Error{fmt.Sprintf("Failed to delete destinations. Error: %s.", err)}
+	}
+
+	if err := store.deleteAllNotificationObjects(map[string]interface{}{"selector": map[string]interface{}{"notification.destinationOrgID": orgID}}); err != nil && err != notFound {
+		return &Error{fmt.Sprintf("Failed to delete notifications. Error: %s.", err)}
+	}
+
```

Comment:
Also need to delete all ACL record associated with this org, see: https://github.com/open-horizon/edge-sync-service/blob/257ebb6fb088d85198d55dc498aaea4a7d7804ef/core/storage/mongoStorage.go#L1863

Reply:
This was not there before, I just noticed that it was added in a [recently merged PR](https://github.com/open-horizon/edge-sync-service/pull/46). I will update it, thanks